js/display.js

```diff
@@ -1610,44 +1611,48 @@ function displaySpellDamage(parent_elem, overallparent_elem, stats, spell, spell
         title_elemavg.appendChild(third_summary);
     }
     else {
-        title_elem.textContent = spell.title;
-        title_elemavg.textContent = spell.title;
+        title_elem.textContent = spell.name;
+        title_elemavg.textContent = spell.name;
     }
 
     parent_elem.append(title_elem);
     overallparent_elem.append(title_elemavg);
 
-    overallparent_elem.append(displayNextCosts(stats, spell, spellIdx));
+    if ('cost' in spell) {
+        overallparent_elem.append(displayNextCosts(stats, spell, spellIdx));
+    }
 
     let critChance = skillPointsToPercentage(stats.get('dex'));
 
-    let save_damages = [];
-
     let part_divavg = document.createElement("p");
     overallparent_elem.append(part_divavg);
 
-    for (let i = 0; i < spell_parts.length; ++i) {
-        const part = spell_parts[i];
-        const damage = damages[i];
+    function _summary(text, val, fmt) {
+        let overallaverageLabel = document.createElement("p");
+        let first = document.createElement("span");
+        let second = document.createElement("span");
+        first.textContent = text;
+        second.textContent = val.toFixed(2);
+        overallaverageLabel.appendChild(first);
+        overallaverageLabel.appendChild(second);
+        second.classList.add(fmt);
+        part_divavg.append(overallaverageLabel);
+    }
+
+    for (let i = 0; i < spell_results.length; ++i) {
+        const damage_info = spell_results[i];
 
         let part_div = document.createElement("p");
         parent_elem.append(part_div);
 
         let subtitle_elem = document.createElement("p");
-        subtitle_elem.textContent = part.subtitle;
+        subtitle_elem.textContent = damage_info.name
         part_div.append(subtitle_elem);
 
-        if (part.type === "damage") {
-            let _results = damage;
-            let totalDamNormal = _results[0];
-            let totalDamCrit = _results[1];
-            let results = _results[2];
-            
-            for (let i = 0; i < 6; ++i) {
-                for (let j in results[i]) {
-                    results[i][j] = results[i][j].toFixed(2);
-                }
-            }
+        if (damage_info.type === "damage") {
```

Comment:
why would it be called damage_info if it wasn't doing damage

Reply:
because me dumb